### PR TITLE
Authorable Cards and Article API Services 

### PR DESCRIPTION
**Pull request by gauthamksn**
Please always provide the Jira Issue your PR is for, as well as test URLs where your change can be observed (before and after):

Jira ID: https://jira.corp.adobe.com/browse/EXLM-8

Test URLs:

- Before: https://main--exlm--adobe-experience-league.hlx.page/en/authorable-cards
- After: https://feature-exlm-8-article-api--exlm--adobe-experience-league.hlx.live/en/authorable-cards-block

### CODE REVIEW

File: scripts/data-service/article-data-service.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+const articleAPIEndpoint = 'https://experienceleague.adobe.com/api/articles/';
```
Comment (Sivaramvt) [inline scripts/data-service/article-data-service.js:1]: Move this to urls.js

File: scripts/browse-card/browse-cards-delegate.js [hunk 2]
```diff
@@ -6,6 +6,8 @@ import BrowseCardsLiveEventsAdaptor from './browse-cards-live-events-adaptor.js'
 import BrowseCardsADLSAdaptor from './browse-cards-adls-adaptor.js';
 import CONTENT_TYPES from './browse-cards-constants.js';
 import { coveoSearchResultsUrl, liveEventsUrl, adlsUrl } from '../urls.js';
+import ArticleDataService from '../data-service/article-data-service.js';
```
Comment (Sivaramvt) [inline scripts/browse-card/browse-cards-delegate.js:9]: Authored cards is one-off case and we could bake this into the block logic

File: blocks/authorable-card/authorable-card.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+  const contentDiv = document.createElement('div');
+  contentDiv.classList.add('authorable-card-content');
+  links.forEach((link) => {
+    BrowseCardsDelegate.handleArticleDataService(link).then((data) => {
+      const cardDiv = document.createElement('div');
+      buildCard(cardDiv, data);
+      contentDiv.appendChild(cardDiv);
+      decorateIcons(block);
+    });
```
Comment (Sivaramvt) [inline blocks/authorable-card/authorable-card.js:46]: What if network calls fails.

File: blocks/authorable-card/authorable-card.css [hunk 4]
```diff
@@ -0,0 +1,108 @@
+.authorable-card {
```
Comment (Sivaramvt) [inline blocks/authorable-card/authorable-card.css:1]: Call it scripts/browse-cards/browse-cards-block.css

**Comment by Sivaramvt** [comment]
1. move the common browse-cards block style central location
2. Move the adapater to block logic and keep the data service clean.

File: scripts/data-service/article-data-service.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+      description: result.Description,
+      tags: result.Tags,
+      copyLink: result.URL,
+      bookmarkLink: '',
+      viewLink: result.URL,
+      viewLinkText: placeholders[this.viewLinkTextPlaceholder]
+        ? placeholders[this.viewLinkTextPlaceholder]
+        : `View ${this.result.contentType}`,
+    };
+  };
```
Comment (noormd08) [inline scripts/data-service/article-data-service.js:123]: Add optional chaining / null check for result object properties.

File: scripts/data-service/article-data-service.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    if (this.result.contentType === 'Article') {
+      this.viewLinkTextPlaceholder = 'viewLinkArticle';
+    }
+    if (this.result.contentType === 'Documentation') {
+      this.viewLinkTextPlaceholder = 'viewLinkDocumentation';
+    }
+    if (this.result.contentType === 'Event') {
+      this.viewLinkTextPlaceholder = 'viewLinkEvent';
+    }
+  }
```
Comment (noormd08) [inline scripts/data-service/article-data-service.js:85]: Optimize the placeholders key values directly via contentType values.

File: scripts/data-service/article-data-service.js [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    if (this.result.contentType === 'Course') {
+      [, thumbnail] = this.result['Full Meta'].match(/course-thumbnail: (.*)/);
+      return thumbnail ? `https://cdn.experienceleague.adobe.com/thumb/${thumbnail.split('thumb/')[1]}` : '';
+    }
+
+    if (this.result.contentType === 'Tutorial') {
+      const videoUrl = this.result['Full Body'].match(/embedded-video src\s*=\s*['"]?([^'"]*)['"]?/)[1];
+      this.result.videoUrl = videoUrl;
+      this.result.videoId = videoUrl?.split('/')[4];
+      return this.result.videoId ? `https://video.tv.adobe.com/v/${this.result.videoId}?format=jpeg` : '';
```
Comment (noormd08) [inline scripts/data-service/article-data-service.js:98]: Optimize this case. 4th element can create issue in future.

**Review by noormd08 — APPROVED** [review]

**Review by noormd08 — APPROVED** [review]

**Pull request merged**